doc/modules/grid_search.rst

```diff
@@ -167,6 +171,342 @@ variable that is log-uniformly distributed between ``1e0`` and ``1e3``::
+   :align: center
+
+The ``ratio`` parameter controls the rate at which the resources grow, and
+the rate at which the number of candidate decreases (more details in
+:ref:`amount_of_resource_and_number_of_candidates`)
+
+.. topic:: Examples:
+
+    * :ref:`sphx_glr_auto_examples_model_selection_plot_successive_halving_heatmap.py`
+    * :ref:`sphx_glr_auto_examples_model_selection_plot_successive_halving_iterations.py`
+
+Choosing ``min_resources`` and the number of candidates
+-------------------------------------------------------
+
+Beside ``ratio``, the two main parameters that influence the behaviour of a
+successive halving search are the ``min_resources`` parameter, and the
+number of candidates (or parameter combinations) that are evaluated.
+``min_resources`` is the amount of resources allocated at the first
+iteration for each candidate. The number of candidates is specified directly
+in :class:`HalvingRandomSearchCV`, and is determined from the ``param_grid``
+parameter of :class:`HalvingGridSearchCV`.
+
+Consider a case where the resource is the number of samples, and where we
+have 1000 samples. With ``min_resources=10`` and ``ratio=2`` we are able to
+run 7 iterations, with the following number of samples: ``[10, 20, 40, 80,
+160, 320, 640]``.
+
+If we start with a high number of candidates, we might end up with a lot of
+candidates at the last iteration. On the other hand if we start with a small
+number of candidates, the last iteration might use less than 640 samples
+which is a waste of resources.
+
+In the case of :class:`HalvingRandomSearchCV`, the number of candidates is set
+by default such that the maximum amount of resources is used at the last
+iteration.
+
+Changing the value of ``min_resources`` will impact the number of possible
+iterations, and as a result will also have an effect on the ideal number of
+candidates.
+
+Another consideration when choosing ``min_resources`` is whether or not it
+is easy to discriminate between good and bad candidates with a small amount
+of resources. For example, if you need a lot of samples to distinguish
+between good and bad parameters, a high ``min_resources`` is recommended. On
+the other hand if the distinction is clear even with a small amount of
+samples, then a small ``min_resources`` may be preferable since it would
+speed up the computation.
+
+Notice in the example above that the last iteration does not use the maximum
+amount of resources available: 1000 samples are available, yet only 640 are
+used, at most. By default, ``min_resources`` is set to a specific value such
+that the last iteration uses as many samples as possible. Please see
+:ref:`exhausting_the_resources` for details.
+
+.. _amount_of_resource_and_number_of_candidates:
+
+Amount of resource and number of candidates at each iteration
+-------------------------------------------------------------
+
+The amount of resources ``resource_iter`` (e.g. the number of samples)
```

Comment:
So that would be 

> The amount of resources at each iteration _i.e._ ``resource_iter`` (e.g. the number of samples)  allocated for each candidate at iteration ``i`` is controlled by the
parameters ``ratio`` and ``min_resources`` as follows::

Honestly, that doesn't read well to me. But I can see how the current one can be confusing, I'll try to come up with something else